pkg/driver/node/mounter/pod_mounter.go

```diff
@@ -77,14 +76,12 @@ func (pm *PodMounter) Mount(ctx context.Context, bucketName string, target strin
 
 	podID := credentialCtx.PodID
 
-	err = pm.verifyOrSetupMountTarget(target)
+	isMountPoint, err := pm.mount.IsMountPoint(target)
```

Comment:
Any reason for changing order between `verifyOrSetupMountTarget()` and `IsMountPoint()`?

The rest LGTM.

Reply:
The `verifyOrSetupMountTarget` was doing a `statx`  syscall and checking the returned error to understand if its needs to create the target folder or re-mount it. Now the idea is to move `statx` call to `IsMountPoint` and if it returns an error just propagate it, so `verifyOrSetupMountTarget` can still handle any returned error from `statx`. `IsMountPoint` was doing regular `stat` syscall before and was only retuning error on "not found" error. 

I realized I forgot to use `statx` in `IsMountPoint`, will do it now. And I can also change the name of `IsMountPoint` to `CheckMountPoint` maybe, as now it will report the errors that `verifyOrSetupMountTarget` will fix instead of simply checking whether the target is mount point.

Reply:
Addressed with https://github.com/awslabs/mountpoint-s3-csi-driver/pull/444/commits/0bdc09aaed2a3954a7c91bd2ab992db6c6c15957 also updated doc comments on `CheckMountPoint`.